src/common/lint/add-lint-configs.ts

```diff
@@ -1,11 +1,42 @@
+import type {Linter} from 'eslint'
+import {JsonFile} from 'projen'
 import {NodeProject} from 'projen/lib/javascript'
+import {deepMerge} from 'projen/lib/util'
+import {eslintConfigFormatting} from './configs/eslint-config-formatting'
+import {eslintConfigQuality} from './configs/eslint-config-quality'
+import {prettierConfig} from './configs/prettier'
 
-export const addLintConfigs = (project: NodeProject, extraEslintConfigs: Array<string> = []): void => {
-  project.package.addField('prettier', '@ottofeller/prettier-config-ofmt')
+const arrayProperties = ['extends', 'overrides', 'plugins'] as const
+type LinterConfigArrayProperties = Pick<Linter.Config, typeof arrayProperties[number]>
 
+export const addLintConfigs = (project: NodeProject, extraEslintConfigs: Array<Linter.Config>): void => {
```

Comment:
@quesabe can we use a single `params` object? So that we get named params in all calls of `addLintConfigs`.

Reply:
Yeah, we can. I'm just more used to the use of multi-arg functions unless there are too many params :)
Let's switch to the single object argument.